wire-library/internal-plugins/src/main/kotlin/com/squareup/gradle/InternalPublishingPlugin.kt

```diff
@@ -67,6 +67,16 @@ class InternalPublishingPlugin : Plugin<Project> {
 
     project.extensions.getByType<PublishingExtension>().apply {
       publications {
+        project.afterEvaluate {
+          // Multiplatform projects get publications automatically, but JVM-only do not. Add 'em.
+          if (this@publications.isEmpty()) {
```

Comment:
This is kinda backwards. You should be reacting to the application of the 'java-library' plugin or something and then creating the publications. And then in afterEvaluate you can query the presence of a Kotlin plugin as to whether to call `java { withSourcesJar() }` or manually attach the Kotlin sources jar.

But... see thing about using the plugin.

Reply:
Good call.